File: mario.py
import cs50

# Prompt user for pyramid's height
while True:
    height = cs50.get_int("Height: ")

    # Handle EOF
    if height is None:
        print()
        exit(0)

    # Validate height
    elif height >= 1 and height <= 8:
        break

# Iterate over pyramid's rows
for i in range(height):

    # Print spaces
    print(" " * (height - i - 1), end="")

    # print hashes
    print("#" * (i + 1))

File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // prompt user for pyramids' height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // iterate over pyramid's rows
    for (int i = 0; i < height; i++)
    {
        // number of spaces to print
        int spaces = height - 1 - i;

        // print spaces
        for (int j = 0; j < spaces; j++)
        {
            printf(" ");
        }

        // print lefthand pyramid
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // print spaces between pyramids
        printf("  ");

        // print righthand pyramid
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // print newline
        printf("\n");
    }
}
